src/stan-users-guide/gaussian-processes.Rmd

```diff
@@ -586,7 +586,72 @@ model {
   y ~ normal(f, sigma);
 }
 ```
+### Noise in Gaussian Process Regression Models
 
+In this section, we'll consider the function below, for no other reason than it has a nice non-linear shape that we can approximate nicely with Gaussian Process regression using a squared exponential kernel. We won't specify the noise immediately, since we'll consider a few different situations.
+
+$$
+f(x_i) = sin(\frac{x_i}{2\pi}) + sin(\frac{.7 x_i}{2\pi})
+$$
+and the observed data, with noise, is then:
+
+$$
+y_i = f(x_i) + \epsilon_i
+$$
+The function $f(x_i)$, without noise for $x=1,2,3,...,100$, looks like this
+
+//////////////GRAPHIC
+
+Previously, we have only considered a noiseless gaussian process. That is, we've fixed $\delta$ to a small constant not less than zero. In gaussian process regression, an identically distributed independent noise is equivalent to having $\delta$, for example, be normally distributed. Specifically, we can set
+
+$$
+\delta \sim normal(0, 1)_+
+$$
+In Stan, we can simulate a GP model with noise by declaring $\delta$ as a parameter and setting $\delta$ to be normally distributed, as above. For a Gaussian likelihood, function, a Stan snippet would be
+
+```
+parameters {
+  ...
+  real<lower=0> delta;
+}
+transformed parameters {
+  matrix[N, N] L_K;
+  {
+    matrix[N, N] K = gp_exp_quad_cov(x, sigma, l);
+    K = add_diag(K, square(delta));
+    L_K = cholesky_decompose(K);
+  }
+}
+model {
+  sigma ~ normal(0, 3);
+  l ~ normal(1, 3);
+  
+  delta ~ normal(0, 1);
+  
+  y ~ multi_normal_cholesky(mu, L_K);
+}
+```
+
+Here's what a Gaussian Process regression fit looks like when the true noise standard deviation is 1:
+
+//////////////GRAPHIC
+
+#### Using a Hierarchical Prior to Estimate Noise from the Data
+
+Unfortunately, this is not always the case, and instead, we could have noise for which that standard deviation is not exactly 1. Here's an example of a fit on simulated data when the noise model is misspecified:
+
+//////////////GRAPHIC
+
+In this case, we can use a slightly more sophisticated method and estimate the noise for the Gaussian Process regression model hierarchically. To do this, we assign a simple hyper prior to the noise of the Gaussian Process. Since the prior has a prior, we call this a hierarchy, and hence a hierarchical prior. Probabilistically, we define the noise model as follows:
+
+$$
+\delta_0 \sim normal_+(0, 100)\\
+\delta \sim normal_+(0, \delta_0)
+$$
+
+We set the prior for the noise more diffuse, and then the noise for the Gaussian Process is estimated from the data. Beware, however, since the hyper prior is more diffuse and the true noise is likely less diffuse, this sub-model has potential to funnel and sample slowly. Here's a statistical graphic of the exact same model, and hence the same Stan code, except when the true noise standard deviation of the simulated data is .5 and 2 respectively. The graphic where seeing is the in sample posterior predictive interval of $\tilde{y}$.
```

Comment:
As delta is scalar, there is no real hierarchy here. This is just a wider prior on delta. Hierachical model would make sense only if there would be many deltas. This section can be removed.